File: packages/runtime-php/src/OAuth2Config.php
<?php

declare(strict_types=1);

namespace Besdk\Runtime;

/** Configuration for an OAuth2 token source (SPEC.md §3.1.6). */
final class OAuth2Config
{
    /**
     * @param list<string> $scopes
     */
    public function __construct(
        public readonly string $tokenUrl,
        public readonly ?string $clientId = null,
        public readonly ?string $clientSecret = null,
        public readonly ?string $refreshToken = null,
        public readonly array $scopes = [],
    ) {}
}

File: packages/runtime-java/src/main/java/graft/runtime/OAuth2Config.java
package graft.runtime;

import java.util.List;

/**
 * Configuration for an OAuth2 token source (SPEC.md §3.1.6).
 *
 * <p>A record with nullable fields rather than a builder: it has five fields, all of which a caller
 * either knows or does not, and no combination is ambiguous.
 */
public record OAuth2Config(
    String tokenUrl,
    String clientId,
    String clientSecret,
    String refreshToken,
    List<String> scopes) {

  public OAuth2Config {
    scopes = scopes == null ? List.of() : List.copyOf(scopes);
  }
}
